pallets/proofs-dealer/src/lib.rs

```diff
@@ -386,6 +386,7 @@ pub mod pallet {
         ProofAccepted {
             provider: ProviderIdFor<T>,
             proof: Proof<T>,
+            last_tick_proof: BlockNumberFor<T>,
```

Comment:
I would rename this to `last_tick_proven` and set it to the last tick for which this `provider` submitted a proof. That tick is not the `current_tick` when the `provider` submits a proof, it's in fact a previous tick that in `do_submit_proof` is under the variable name `challenges_tick`.